### DISCUSSION TITLE
Override IncludeIf Attribute

### DISCUSSION THREAD

**Discussion by Mohsnb** [paolo-rossi/delphi-neon#64] [Ideas]
Hello and thanks for this project.
Please add this future can set include attribute in config. i changed this way:
in `Persistence` unit add this function to `INeonConfiguration` :
`function SetInclude(AValue: IncludeIf): INeonConfiguration;`

In `TNeonConfiguration` add this members:
```
private  
    FSetedInclude: Boolean;
    FSetedIncludeValue: IncludeIf;
public
    function SetInclude(AValue: IncludeIf): INeonConfiguration;
    property SetedInclude: Boolean read FSetedInclude write FSetedInclude default False;
    property SetedIncludeValue: IncludeIf read FSetedIncludeValue write FSetedIncludeValue;
```

And :
```
constructor TNeonConfiguration.Create;
begin
  FSetedInclude:= False;
...

function TNeonConfiguration.SetInclude(AValue: IncludeIf): INeonConfiguration;
begin
  FSetedInclude:= True;
  FSetedIncludeValue:= AValue;
  Result := Self;
end;
```

In `TNeonSerializerJSON.WriteDataMember` in `Neon.Core.Persistence.JSON` unit and before `case AValue.Kind of` add this lines:
```
var
  LSetedInclude: TIncludeValue;
```
```
  if FConfig.SetedInclude then
  begin
    LSetedInclude.Value:= FConfig.SetedIncludeValue;
    ANeonObject.NeonInclude:= LSetedInclude;
  end;
```

Now in config can set like this:
```
LConfig := TNeonConfiguration.Default
    .SetInclude(IncludeIf.Always);
```

**Comment by paolo-rossi** [comment]
@Mohsnb 

sorry it took me so long to respond, I tend to read the discussions not so often (yeah I know, my bad)... anyway...

This is a very good suggestion, in order to set a customized default for the IncludeIf value.

If you don't want to create a PR I will add the code myself.

Thanks for the suggestion,
Paolo.
